Feature/kas 3026 aanpassen pubstatus

TaneeG opened the pull request:
jira: https://kanselarij.atlassian.net/browse/KAS-3026
backend: https://github.com/kanselarij-vlaanderen/app-kaleidos/pull/250
jenkins: http://kal-kastaar.s.redpencil.io:8080/job/kaleidos/job/custom_frontend_backend_combination/154/

Commit:
KAS-3026 Added new statusses & remove old code from sidebar

Commit:
KAS-3026 Added statusSelection modal

Commit:
Merge branch 'development' into feature/kas-3026_Aanpassen_pubstatus

Commit:
KAS-3026 Fix loading problem with status & statuschange

Commit:
KAS-3026 Add blue color to circles

erikap committed:
Merge branch 'development' into feature/kas-3026_Aanpassen_pubstatus

erikap committed:
Fix publication-status to publications relation name

erikap committed:
Add newlines between get-functions

erikap commented on app/components/publications/publication-sidebar.hbs:
```diff
@@ -93,7 +93,7 @@
 
         {{! Spoedprocedure }}
-        {{#if this.publicationStatus.isPending}}
+        {{#if this.publicationStatus.isStarted}}
```
Deze aanpassing is volgens mij te kort door de bocht. Overal waar voorheen iets getoond werd bij status `isPending` moet dit volgens mij nu getoond worden in elke status die niet gepauzeerd of finaal is. De vroegere status `isPending` is nu opgesplitst in verschillende andere statussen (proef in, rappel proef, ...)

Ik zou daarom een getter `isPending` (of `isOngoing`) toevoegen op publication-status data model dat `true` geeft bij alle statussen behalve `isPaused`, `isPublished` of `isWithdrawn`.

erikap commented on app/pods/publications/publication/template.hbs:
```diff
@@ -6,25 +6,40 @@
+              {{on "click" this.openStatusSelector}}
+            >
+              {{t "edit-status"}}
+            </Auk::ButtonLink>
```
Kan je de weergave en editeren van de status in een aparte component steken? Dat gaat volgens mij de route/controller vereenvoudigen.

erikap reviewed (changes requested):
Bij het (proberen) testen van deze story ben ik wat issues tegengekomen met het opslaan van de relatie tussen publication-flow en publication-status-change. Die is niet geintroduceerd door deze story, maar ik snap niet goed hoe je dit zelf hebt kunnen testen :confused: 

1. Bij het aanmaken van een nieuwe publicatie wordt er wel een publication-status-change resource gemaakt, maar de relatie met de publication-flow wordt niet correct opgeslagen (oorzaak: verkeerde relatienaam in https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/feature/kas-3026_Aanpassen_pubstatus/app/services/publication-service.js#L91). Hierdoor krijg ik een error wanneer ik de status probeer te wijzigen van een nieuwe publicatie.
2. De relatie tussen publication-flow en publication-status-change is one-to-one. Het is dus de bedoeling om enkel voor de laatste wijziging de publication-status-change bij te houden en niet voor alle historische wijzigingen. Op dit moment worden er enkel nieuwe publication-status-change resources toegevoegd zonder de vorige te verwijderen. (de backend zal nu een van die publication-status-change resources kiezen om terug te geven voor de relatie, maar wanneer dit niet de laatste is, wordt een verkeerde datum getoond in de frontend)

erikap committed:
Merge branch 'development' into feature/kas-3026_Aanpassen_pubstatus

Commit:
KAS-3026 Add pending status

Commit:
KAS-3026 Add Separate component for viewing pub status

Commit:
KAS-3026 Fix faulty status change link

TaneeG commented:
@erikap Deze is terug klaar voor review.
Heb die faulty link ook hersteld. 

Design is nog niet klaar voor volgens prototype dit gaat @brenner-company later opnemen

erikap committed:
Merge branch 'development' into feature/kas-3026_Aanpassen_pubstatus

erikap committed:
Set default publication change date to today

erikap committed:
Use getter instead of tracked property for static list

erikap committed:
Fix loading state of save publication status button in modal

erikap committed:
Move meaningless actions down

erikap committed:
Add derived isFinal state on publication-status model

erikap committed:
Create new publication-status-change activity on each update

erikap committed:
Remove publication withdrawl confirmation

Since there is no autosave anymore on the status update of a
publication, an additional confirmation for update to the withdrawl
state is no longer needed

erikap committed:
Convert publication status save to an ember-concurrency task

erikap committed:
Update publication closing date to selected date instead of current date

erikap committed:
Rename publication-status-display to publication-status-pill

erikap commented:
Voornaamste wijziging die ik nog gedaan heb, is het verwijderen van de extra confirmatie bij het annuleren van een publicatie. Die was vroeger toegevoegd omdat in de sidebar het aanpassen van de status automatisch opgeslagen werd. Dat is nu niet meer het geval. De gebruiker zit al in een modal waar hij expliciet op 'Opslaan' moet klikken om de change te bevestigen. Die extra confirmatie is dus overbodig.

erikap committed:
Disable save as long as no new status is selected for publication

erikap committed:
Rename toPublishStatus to initialStatus on publication creation

erikap committed:
Disable tests on publication-status changes
